### PR TITLE
merge AccountScriptingInterface into GlobalServicesScriptingInterface

### PR DESCRIPTION
**Pull request by dback2**
- move isLoggedIn, logOut, loggedIn (including associated member flag variable), and checkAndSignalForAccessToken functions from Account to Global Services API and wrap existing calls from Account API to their associated Global Services API functions
- add loggedInChanged to Global Services API as well as a wrapper function onUsernameChanged to mimic the loggedInChanged behavior from Account API as well as to call myUsernameChanged signal as it was directly doing in Global Services API via AccountManager::usernameChanged listener before
- remove checkAndSignalForAccessToken, usernameChanged, loggedInChanged, and onUsernameChanged from Account API as nothing was using them there
- point Account context to GlobalServicesScriptingInterface instead of AccountScriptingInterface to effectively sever all direct uses of Account API

In later PRs:
-Delete AccountScriptingInterface class completely.
-Rename GlobalServicesScriptingInterface to AccountScriptingInterface (we have decided since Account is more used than GlobalServices that Account naming will be used instead for the long term).
-Remove the GlobalServices contexts (at a set date after advanced notice of removal).

Test Plan:

https://highfidelity.testrail.net/index.php?/cases/view/7536 (Desktop)
https://highfidelity.testrail.net/index.php?/cases/view/7535 (HMD)

These changes are only a behind the scenes refactor and not a functionality change. This should cause no visible differences in functionality for anything account related such as logging in/out, checking login status, accessing username, using Marketplace, etc.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4400/HighFidelity-Beta-PR11669-cffcf2d8df42c9c7e22c5fa75c08f27d74cd9e86.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4400/HighFidelity-Beta-PR11669-cffcf2d8df42c9c7e22c5fa75c08f27d74cd9e86.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4403/HighFidelity-Beta-PR11669-c8a0f9761a5670c370b1f6bb331f5718b9b64637.exe

**Comment by ZappoMan** [comment]
One side effect of this change is that all the `Q_INVOKABLE` and `Q_PROPERTY` members of `GlobalServicesScriptingInterface ` will not be exposed to JS as methods of `Account` is that a good thing?

**Comment by dback2** [comment]
Why would they not exposed if Account context points to GlobalServicesScriptingInterface?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5374/HighFidelity-Beta-PR11669-3873a8f91ef4be8fbee47111e5e9a4dbef66e46b.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5383/HighFidelity-Beta-PR11669-3873a8f91ef4be8fbee47111e5e9a4dbef66e46b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5383/HighFidelity-Beta-PR11669-3873a8f91ef4be8fbee47111e5e9a4dbef66e46b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5426/HighFidelity-Beta-PR11669-5e1064800a1e5eec5dff0bb84eec8a868d53ded5.exe

**Comment by dback2** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5434/HighFidelity-Beta-PR11669-5e1064800a1e5eec5dff0bb84eec8a868d53ded5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5434/HighFidelity-Beta-PR11669-5e1064800a1e5eec5dff0bb84eec8a868d53ded5.dmg

**Comment by Nex-Pro** [comment]
Testing Win 10

**Comment by Nex-Pro** [comment]
Stopped part way through test rail. Unable to connect to dev-content.

Also IMO test rail needs breaking into HMD and desktop.  It also has two links to other suites of tests embedded in it. @dback2  Was fine appart from some minor inaccuracies in the expected results up to that point.

https://highfidelity.testrail.net/index.php?/tests/view/24288&group_by=cases:section_id&group_order=asc&group_id=1252 it is marked as pass but is blocked.

**Comment by dback2** [comment]
Split test plans into HMD and Desktop.
Fixed links to the 2 external test plans to regress.
Adjusted test plan to use hifiqa-rc instead of Avatar Island.
Should be ready to test again.

File: interface/src/scripting/GlobalServicesScriptingInterface.cpp [hunk 1]
```diff
@@ -45,8 +48,29 @@ GlobalServicesScriptingInterface* GlobalServicesScriptingInterface::getInstance(
...
 }
 
-const QString& GlobalServicesScriptingInterface::getUsername() const {
-    return DependencyManager::get<AccountManager>()->getAccountInfo().getUsername();
+const QString GlobalServicesScriptingInterface::getUsername() const {
+    auto accountManager = DependencyManager::get<AccountManager>();
+    if (accountManager->isLoggedIn()) {
+        return accountManager->getAccountInfo().getUsername();
+    }
+    else {
```
Comment (thoys) [inline interface/src/scripting/GlobalServicesScriptingInterface.cpp]: small style issue: `} else {`
https://docs.highfidelity.com/build-guide/coding-standards#4-1-5-the-if-else-class-of-statements-should-have-the-following

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Failed
### [Test Run](https://highfidelity.testrail.net/index.php?/runs/view/1746&group_by=cases:section_id&group_order=asc)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5769/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5769/HighFidelity-Beta-PR11669-be736373f29b77e24429eed651f2bd2b75f82730.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5769/HighFidelity-Beta-PR11669-be736373f29b77e24429eed651f2bd2b75f82730.dmg

**Comment by dback2** [comment]
I think we probably need to hold for hifiqa-rc update as it's the only dev domain that I know of where avatar wearable checkout can currently be tested.

**Comment by dback2** [comment]
Checkout zone added to dev-content at around 80,-10,10 and test plan adjusted.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5845/app_11669_PR.apk

**Comment by dback2** [comment]
Not sure what happened to the original test plans but instead of going to hifiqa-rc you can now go to dev-content/80,-10,10 and use the checkout zone there to test checkout of wearables instead.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5845/HighFidelity-Beta-PR11669-ab5b238aea1be0813a25e095359a845527a85044.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5845/HighFidelity-Beta-PR11669-ab5b238aea1be0813a25e095359a845527a85044.dmg

**Comment by sbennett77** [comment]
Currently testing

**Comment by sbennett77** [comment]
Testing Passed.

**Comment by dback2** [comment]
Test plan links fixed

**Comment by thoys** [comment]
This will need a re-test, correctly updated test plans are PR description.

**Comment by QAadahlin** [comment]
Currently testing

**Comment by QAadahlin** [comment]
@conklin94122 fails

**Comment by QAadahlin** [comment]
@conklin94122 fails

When you attempt to change the security picture, it hangs in "submitting" limbo and never finishes loading.

**Comment by QAadahlin** [comment]
@conklin94122 - fails
When you attempt to change the security picture, it hangs in "submitting" limbo and never finishes loading.
securitypic
If you press the cancel button in the "change security picture" , then go to "change passphrase", the following image is what you see. 
wallet
When you go to change the security passphrase first, it takes. You can then go to attempt to change your security picture (which again stays in "submitting" limbo) without seeing the latter defect.

**Comment by dback2** [comment]
@QAadahlin @conklin94122 please verify this failure does not exist on master build

**Comment by conklin94122** [comment]
It didn't fail in a different PR build a few minutes ago but I'll check master now to be safe.

**Comment by QAadahlin** [comment]
Testing on latest build

**Comment by QAadahlin** [comment]
@conklin94122 #11669 PASSED on the latest build (build #7616).

**Comment by dback2** [comment]
PR notes for 7616 specifically call out the above issue

**Comment by conklin94122** [comment]
@dback2 Can you rebase this PR against current master and we can re-retest?

**Comment by dback2** [comment]
@conklin94122 done build incoming :)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5927/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5927/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5927/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5928/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5928/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5928/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.dmg

**Comment by QAadahlin** [comment]
@conklin94122 Re-testing build

**Comment by QAadahlin** [comment]
@conklin94122 BLOCKED due to inability to access dev-content/80,-10,10, hifi://empty, or hifi://hifiqa-rc. Previous bug has been fixed and passed.

**Comment by dback2** [comment]
@conklin94122 @QAadahlin Try dev-content/80,-10,10 again you should now have access

**Comment by conklin94122** [comment]
Thanks much, @dback2

**Comment by QAadahlin** [comment]
@conklin94122 Currently retesting

**Comment by QAadahlin** [comment]
@conklin94122 Fails. You cannot edit, remove, or purchase the wearable object. When you select the object. The only way to remove the item was to leave the domain. 
screen shot 2018-01-04 at 1 04 48 pm copy

**Comment by dback2** [comment]
Rebasing for protocol change - ready for retest on new build

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5949/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5949/HighFidelity-Beta-PR11669-81f3590954d26350da0208c15e49d0fce1c2c75d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5949/HighFidelity-Beta-PR11669-81f3590954d26350da0208c15e49d0fce1c2c75d.dmg

**Comment by QAadahlin** [comment]
@conklin94122 Re-testing with new build

**Comment by QAadahlin** [comment]
@conklin94122 No longer testing #11669

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5968/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5968/HighFidelity-Beta-PR11669-ac8b8da7b1ab0c0dc3926b696f863f25e7a4c76e.exe

**Comment by conklin94122** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5986/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5986/HighFidelity-Beta-PR11669-ac8b8da7b1ab0c0dc3926b696f863f25e7a4c76e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5986/HighFidelity-Beta-PR11669-ac8b8da7b1ab0c0dc3926b696f863f25e7a4c76e.dmg

**Comment by QAadahlin** [comment]
@conklin94122 Currently re-testing new build

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by QAadahlin** [comment]
@conklin94122 Testing complete. PASSED.

**Comment by conklin94122** [comment]
QA approved but it has conflicts. Please let me know if the changes invalidate testing and we need to retest, @dback2

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6013/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6014/app_11669_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6014/HighFidelity-Beta-PR11669-76f278f8998bdb70f1916047ae157fb6205595c4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6014/HighFidelity-Beta-PR11669-76f278f8998bdb70f1916047ae157fb6205595c4.exe

**Pull request merged**